lib/koala/api/graph_batch_api.rb

```diff
@@ -53,7 +53,15 @@ def execute(http_options = {})
           end
 
           original_api.graph_call("/", args, "post", http_options) do |response|
-            raise bad_response if response.nil?
+            raise bad_response("Facebook returned an empty body") if response.nil?
+
+            # when http_component is set we receive Koala::Http_service response object
+            # from graph_call so this needs to be parsed
+            # as generate_results method handles only JSON response
+            if http_options[:http_component]
```

Comment:
here we need to add a check for :response because if you set http_component = header or status it will fail see https://github.com/pri1012/koala/blob/c3279139fab69d25d017cb9b83d1de9e18fd5018/lib/koala/api.rb#L57